File: Python/Level 3/네트워크 (DFS).py
visit = [False] * 201

def dfs(start,computers):
    visit[start] = True
    
    for i in range(len(computers)):
        if visit[i] == False and computers[start][i] == 1:
            dfs(i,computers)
    
    return start

def solution(n, computers):
    answer = 0
    for i in range(n):
        if visit[i] == False:
            dfs(i,computers)
            answer += 1
    return answer

File: Javascript/Level 3/네트워크(DFS).js
let visit = new Array(201,0);

function dfs(computers,start,n){
    visit[start]=1;
    
    for(let i=0;i<n;i++){
        if(computers[start][i] && visit[i]!==1){
            dfs(computers,i,n);
        }
    }
}

function solution(n, computers) {
    var answer = 0;
    
    for(let i=0;i<computers.length;i++)
        {
            if(visit[i]!==1)
            {
                dfs(computers,i,n);
                answer++;
            }
        }
    
    return answer;
}

// https://programmers.co.kr/learn/courses/30/lessons/43162?language=javascript
// DFS로 연결요소 개수 출력